src/types/abstract_hmm.jl

```diff
@@ -44,20 +57,30 @@ Return a type that can accommodate forward-backward computations for `hmm` on ob
 It is typically a promotion between the element type of the initialization, the element type of the transition matrix, and the type of an observation logdensity evaluated at `obs`.
 """
 function Base.eltype(hmm::AbstractHMM, obs, control)
-    init_type = eltype(initialization(hmm))
-    trans_type = eltype(transition_matrix(hmm, control))
-    dist = obs_distributions(hmm, control, obs)[1]
-    logdensity_type = typeof(logdensityof(dist, obs))
-    return promote_type(init_type, trans_type, logdensity_type)
+	init_type = eltype(initialization(hmm, control))
+    	trans_type = eltype(transition_matrix(hmm, control))
+    	dist = obs_distributions(hmm, control, obs)[1]
+    	logdensity_type = typeof(logdensityof(dist, obs))
+    	return promote_type(init_type, trans_type, logdensity_type)
 end
 
+	
+#--------------------------- 2.3. initialization --------------------------#
 """
     initialization(hmm)
 
 Return the vector of initial state probabilities for `hmm`.
 """
 function initialization end
 
+initialization(hmm::AbstractHMM) = hmm.init
+
+initialization(hmm::AbstractHMM, control) = hmm.init
+	
+initialization(hmm::AbstractHMM, ::Nothing) = hmm.init
```

Comment:
The point of `AbstractHMM` is that is shouldn't be tied to a specific implementation. Here you implicitly assume that the field `init` is present, which may not be the case